Added docs + docstrings (#102)
datajoint/element-array-ephys: docs/src/concepts.md

# Concepts

## Acquisition Tools for Electrophysiology

### Neuropixels Probes

> Neuropixels probes were developed by a collaboration between HHMI Janelia, industry partners, and others (Jun et al., Nature 2017). Since their initial release in October 2018, 300 labs have ordered 1200 probes. Since the rollout of Neuropixels 2.0 in October 2020, IMEC has been shipping 100+ probes monthly (correspondence with Tim Harris).

Reviewer: 1. Please hard wrap
2. Please implement the convention present in electrode-localization for references

Author: Hopefully the wrap worked. I thought VSCode was doing that automatically. 

References will be updated in the next commit

## Key Partnerships

Over the past few years, several labs have developed DataJoint-based data management and processing pipelines for Neuropixels probes. Our team collaborated with several of them during their projects. Additionally, we interviewed these teams to understand their experimental workflow, pipeline design, associated tools, and interfaces. These teams include:

- [International Brain Lab](https://internationalbrainlab.org) - [https://github.com/int-brain-lab/IBL-pipeline](https://github.com/int-brain-lab/IBL-pipeline)

- [Mesoscale Activity Project (HHMI Janelia)](https://github.com/mesoscale-activity-map) - [https://github.com/mesoscale-activity-map/map-ephys](https://github.com/mesoscale-activity-map/map-ephys)

- Moser Group (Norwegian University of Science and Technology) - see [pipeline design](https://moser-pipelines.readthedocs.io/en/latest/ephys/overview.html)

- Andreas Tolias Lab (Baylor College of Medicine)

- BrainCoGs (Princeton Neuroscience Institute)

- Brody Lab (Princeton University)

## Element Architecture

> Each of the DataJoint Elements are a set of tables for common neuroinformatics modalities to organize, preprocess, and analyze data. Each node in the following diagram is either a table in the Element itself or a table that would be connected to the Element.

Reviewer 2: ```suggestion
Each of the DataJoint Elements creates a set of tables for common neuroscience data modalities to organize, preprocess, and analyze data. Each node in the following diagram is a table within the Element or a table connected to the Element.
```

![element-array-ephys diagram](https://raw.githubusercontent.com/datajoint/element-array-ephys/main/images/attached_array_ephys_element_acute.svg)

### `subject` schema ([API docs](https://datajoint.com/docs/elements/element-animal/api/element_animal/subject))
- Although not required, most choose to connect the `Session` table to a `Subject` table.

| Table | Description |
| --- | --- |
| Subject | Basic information of the research subject |

### `session` schema ([API docs](https://datajoint.com/docs/elements/element-session/api/element_session/session_with_datetime))

| Table | Description |
| --- | --- |
| Session | Unique experimental session identifier |

> ### `probe` schema ([API docs](../api/element_array_ephys/probe))

Reviewer: please verify that this link works in local deployment

Author: It is working in local

Tables related to the Neuropixels probe and electrode configuration.

| Table | Description |
| --- | --- |

> | ProbeType | a lookup table specifying the type of Neuropixels probe (e.g. "neuropixels 1.0", "neuropixels 2.0 single-shank"). |

Reviewer: Please follow the convention above, with the description in sentence case

Author: Done

## Element Development

Through our interviews and direct collaboration on the precursor projects, we identified the common motifs to create the [Array Electrophysiology Element](https://github.com/datajoint/element-array-ephys).

Major features of the Array Electrophysiology Element include:

+ Pipeline architecture detailed by:

    + Probe, electrode configuration compatible with Neuropixels probes and generalizable to other types of probes (e.g. tetrodes) - supporting both `chronic` and `acute` probe insertion modes.

    + Probe-insertion, ephys-recordings, LFP extraction, clusterings, curations, sorted units and the associated data (e.g. spikes, waveforms, etc.).
    
    + Store/track/manage different curations of the spike sorting results - supporting both curated clustering and kilosort triggered clustering (i.e., `no_curation`).

+ Ingestion support for data acquired with SpikeGLX and OpenEphys acquisition systems. 
+ Ingestion support for spike sorting outputs from Kilosort.
+ Triggering support for workflow integrated Kilosort processing.
+ Sample data and complete test suite for quality assurance.

> Incorporation of SpikeInterface into the Array Electrophysiology Element will be on DataJoint Elements development roadmap. Dr. Loren Frank has led a development effort of a DataJoint pipeline with SpikeInterface framework and NeurodataWithoutBorders format integrated (https://github.com/LorenFrankLab/nwb_datajoint).

Reviewer: If we say this here, it should go in roadmap below

Author: Changed it, just pending commit.

## Data Export and Publishing

Element Array Electrophysiology supports exporting of all data into standard Neurodata Without Borders (NWB) files. This makes it easy to share files with collaborators and publish results on [DANDI Archive](https://dandiarchive.org/). [NWB](https://www.nwb.org/), as an organization, is dedicated to standardizing data formats and maximizing interoperability across tools for neurophysiology. For more information on uploading NWB files to DANDI within the DataJoint Elements ecosystem, visit our documentation for the DANDI upload feature of [Element Array Electrophysiology](datajoint.com/docs/elements/element-array-ephys/).

## Roadmap

> Further development of this Element is community driven. Upon user requests we will continue adding features to this Element.

Reviewer: @sidhulyalkar - Can you think of an example to add here? i.e. 'features to this Element, such as ...'

Reviewer 3: Can add that we will plan to add functionality to support large (48 hour) neuropixel recordings via an overlapping segmented processing approach.
